frappe/model/document.py

```diff
@@ -1143,6 +1143,20 @@ def add_seen(self, user=None):
 				self.db_set('_seen', json.dumps(_seen), update_modified=False)
 				frappe.local.flags.commit = True
 
+	def add_viewed(self, user=None):
+		'''add log to communication when a user viewes a document'''
+		if not user:
+			user = frappe.session.user
+
+		if self.meta.track_views:
+			frappe.get_doc({
+				"doctype": "View log",
+				"viewed_by": frappe.session.user,
+				"reference_doctype": self.doctype,
+				"reference_name": self.name,
+			}).insert(ignore_permissions=True)
+			frappe.db.commit()
```

Comment:
`frappe.db.commit` is not a good pattern. It may cause unintended commits! Add a commit at the request level.

or rename the method as `add_viewed_and_commit`

Reply:
How can I add a commit at request level?